homeassistant/components/flexmeasures/config_flow.py

```diff
@@ -0,0 +1,239 @@
+        vol.Required("soc_min", description={"suggested_value": 0.001}): float,
+        vol.Required("soc_max", description={"suggested_value": 0.002}): float,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    host, ssl = get_host_and_ssl_from_url(data["url"])
+
+    # Currently used here solely for config validation (i.e. not returned to be stored in the config entry)
+    try:
+        client = FlexMeasuresClient(
+            session=async_get_clientsession(hass),
+            host=host,
+            email=data["username"],
+            password=data["password"],
+            ssl=ssl,
+        )
+    except Exception as exception:
+        raise CannotConnect(exception) from exception
+    try:
+        await client.get_access_token()
+    except Exception as exception:
+        raise InvalidAuth(exception) from exception
+
+    # Return info that you want to store in the config entry.
+    return {"title": "FlexMeasures"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for FlexMeasures integration."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        # Support a single FlexMeasures configuration only
+        await self.async_set_unique_id(DOMAIN)
+        self._abort_if_unique_id_configured()
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect as exception:
+            errors["base"] = "cannot_connect"
+
+            for field in ("url", "email", "password"):
+                if field in str(exception):
+                    errors[field] = str(exception)
+
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        # Show form again, showing captured errors
+        # still do invalid_auth validation error is not yet shown properly
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
Please look into using `SchemaOptionsFlowHandler` it helps you inject the current settings in an options flow etc